radicle-httpd/src/api.rs

```diff
@@ -0,0 +1,161 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+use std::time::Duration;
+
+use axum::body::{Body, BoxBody};
+use axum::http::header::{AUTHORIZATION, CONTENT_TYPE};
+use axum::http::Method;
+use axum::response::{IntoResponse, Json};
+use axum::routing::get;
+use axum::{Extension, Router};
+use hyper::http::{Request, Response};
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use tokio::sync::RwLock;
+use tower_http::cors::{self, CorsLayer};
+use tower_http::trace::TraceLayer;
+use tracing::Span;
+
+use radicle::identity::{Doc, Id};
+use radicle::storage::{ReadRepository, WriteStorage};
+use radicle::Profile;
+
+mod auth;
+mod axum_extra;
+mod error;
+mod v1;
+
+pub const VERSION: &str = env!("CARGO_PKG_VERSION");
+
+/// Identifier for sessions
+type SessionId = String;
+
+#[derive(Clone)]
+pub struct Context {
+    profile: Arc<Profile>,
+    sessions: Arc<RwLock<HashMap<SessionId, auth::AuthState>>>,
+}
+
+impl Context {
+    pub fn new(profile: Arc<Profile>) -> Self {
+        Self {
+            profile,
+            sessions: Default::default(),
+        }
+    }
+
+    pub fn project_info(&self, id: Id) -> Result<project::Info, error::Error> {
+        use radicle::cob::issue::Issues;
```

Comment:
```suggestion
        use radicle::cob::issue::Issues;
        
```

Reply:
ah, this was temporarily placed here. will move it up.